### PR TITLE
open chromatin function updated

### PR DESCRIPTION
**Pull request by kkupkova**
Following changes were made:

1) Cell-tissue metadata were updated to make them work also with new mm10 matrix.
2) Example matrix was updated and now contains values after new normalization.
3) Calc function now calculates also box plot statistics - these are not calculated as part of plot function anymore. Calc function now returns a list with two tables in them - i) matrix with cell specific signal values, ii) matrix with box plot stats for a given cell type (upper whisker, upper hinge, median, lower whisker, lower hinge).
4) Updated tests, since output of calc function is a list and data.table anymore.

### CODE REVIEW

**Comment by nsheff** [comment]
Hi @kkupkova, in the future, please refer to the issues you are addressing in commits if you can. If not or in addition, then at least add them to the comments here on the pull request so that the issues are linked to the solutions.

**Comment by nsheff** [comment]
Getting this message:
```
   plotOpenSignal: no visible binding for global variable ‘tissueType’
   Undefined global functions or variables:
     tissueType
```

**Comment by nsheff** [comment]
Did you run R CMD check and R CMD build and install, etc.? You should always run all these tests. Here's how I do it:

```
packageName="GenomicDistributions"

packageFolder=paste0("~/code/", packageName) # local path
devtools::document(packageFolder)
devtools::install(packageFolder, upgrade="never")
devtools::test(packageFolder)
devtools::run_examples(packageFolder)
devtools::check(packageFolder)
devtools::build_vignettes(packageFolder)
```

**Comment by nsheff** [comment]
this appears to be still the old data: http://big.databio.org/GenomicDistributionsData/ 
where is the new data? Can you put it there? 

Please ensure that the long vignette works as you expect

https://github.com/databio/GenomicDistributions/tree/master/long_vignettes

right now this appears to still be using the old data.

**Review by nsheff — CHANGES_REQUESTED** [review]

**Comment by kkupkova** [comment]
> Hi @kkupkova, in the future, please refer to the issues you are addressing in commits if you can. If not or in addition, then at least add them to the comments here on the pull request so that the issues are linked to the solutions.

Ok, I will do it next time and I will clean up the issue tracker.

**Comment by kkupkova** [comment]
> Getting this message:
> 
> ```
>    plotOpenSignal: no visible binding for global variable ‘tissueType’
>    Undefined global functions or variables:
>      tissueType
> ```

Where do you get this? Is it possible, that you are using the old metadata associated with the matrices? Because I checked the cellTypeMetadata (loaded on line 156: `cellTypeMetadata = getReferenceData("","cellTypeMetadata")`) contains the correct column name and I am not getting any error.

**Comment by kkupkova** [comment]
> http://big.databio.org/GenomicDistributionsData/

Sorry, I had no idea about this archive. Should I replace it, or increase the version and keep both?

**Comment by nsheff** [comment]
> Where do you get this? Is it possible, that you are using the old metadata associated with the matrices? Because I checked the cellTypeMetadata (loaded on line 156: cellTypeMetadata = getReferenceData("","cellTypeMetadata")) contains the correct column name and I am not getting any error.

it's somewhere in the R check process.

**Comment by nsheff** [comment]
> Sorry, I had no idea about this archive. Should I replace it, or increase the version and keep both?

I guess it's always better to increment versions

File: R/specificity-plots.R [hunk 1]
```diff
@@ -206,40 +230,41 @@ plotOpenSignal = function(signalMatrix,
...
       theme_bw() +
       theme(axis.text.x = element_text(angle = 90, hjust = 1),
             text = element_text(size=10)) +
       xlab("") +
       ylab("normalized signal") + 
       scale_x_discrete(labels = myLabels$spaceLabel) +
-      scale_fill_manual(values=colorScheme) + 
-      scale_color_manual(values=colorScheme) +
-      ylim(minBoxLimit, maxBoxLimit)
+      scale_fill_manual(values = colorScheme) +
```
[the remark this answers is not in the document]
Comment (kkupkova) [reply R/specificity-plots.R:257]: I corrected that through both of the functions. Will be included in the next push.

File: R/specificity-plots.R [hunk 2]
```diff
@@ -27,9 +30,22 @@ calcOpenSignal = function(query, cellMatrix){
     if(is.null(nameList)) {
       nameList = seq_along(query) # Fallback to sequential numbers
     }
-    # Append names
-    xb = rbindlist(x)
-    xb$name = rep(nameList, vapply(x, nrow, integer(1)))
+    # Extract signal matrices and boxplot matrices
+    # rbind them and add a name column
+    signalList = lapply(x, `[[`, 1)
```
[the remark this answers is not in the document]
Comment (kkupkova) [reply R/specificity-plots.R:35]: 1. calcOpenSignal does return named list - for plotting purposes I just wanted it to return list with two matrices within them, which would have "name" column in them. So this step is just to create a matrix from list of lists, so I don't have to completely rewrite the plotting function. This is the only solution I fount to access all the first elements in the lists and all the second elements of lists - then combine all firsts into a matrix and all seconds into another matrix. The output of the function is, however, a named list. 

2. I can change it, but this is, how it's done in all of the other calc functions - exactly like this. I was thinking that I would then keep it, so the function has the same structure like the others.

**Comment by kkupkova** [comment]
I think that I managed to address all the comments. 
I updated the long vignette - now it's using the updated GenomicDistributionData package (v_0.0.2).  
#107 should be whole resolved now - I split the main function into smaller functions to use maximum 50 lines per function.
#122 - updated in README (also in master branch) - not updated in vignettes yet.

**Comment by nsheff** [comment]
Ok. I will go ahead and merge this and then give everything a test run next week.

**Pull request merged**